### PR TITLE
Return error when create message stream in QueryNode

### LINKED ISSUE #15312: [Bug]: querynode ignore error when create msgstream
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

when failed to create msgstream, the error should be handled. Because msgstream.client== nil at this time, so will panic when creating the consumer
```
func (q *queryNodeFlowGraph) newDmInputNode(ctx context.Context, factory msgstream.Factory) *flowgraph.InputNode {
	insertStream, err := factory.NewTtMsgStream(ctx)
	if err != nil {
		log.Warn(err.Error())
	} else {
		q.dmlStream = insertStream
	}

	maxQueueLength := Params.QueryNodeCfg.FlowGraphMaxQueueLength
	maxParallelism := Params.QueryNodeCfg.FlowGraphMaxParallelism

	node := flowgraph.NewInputNode(insertStream, "dmlInputNode", maxQueueLength, maxParallelism)
	return node
}
```

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by bigsheeper**
Signed-off-by: bigsheeper <yihao.dai@zilliz.com>

resolves: #15312

### CODE REVIEW

File: internal/querynode/task.go [hunk 1]
```diff
@@ -377,15 +377,22 @@ func (w *watchDmChannelsTask) Execute(ctx context.Context) error {
 	)
 
 	// add flow graph
-	channel2FlowGraph := w.node.dataSyncService.addFlowGraphsForDMLChannels(collectionID, vChannels)
+	channel2FlowGraph, err := w.node.dataSyncService.addFlowGraphsForDMLChannels(collectionID, vChannels)
+	if err != nil {
+		log.Warn("watchDMChannel, add flowGraph for dmChannels failed", zap.Int64("collectionID", collectionID), zap.Strings("vChannels", vChannels), zap.Error(err))
```
Comment (xige-16) [inline internal/querynode/task.go:382]: when err != nil, the returned channel2FlowGraph is nil. This process of cleaning up the Flow graph is invalid

File: internal/querynode/task.go [hunk 2]
```diff
@@ -524,13 +528,17 @@ func (w *watchDeltaChannelsTask) Execute(ctx context.Context) error {
...
-	channel2FlowGraph := w.node.dataSyncService.addFlowGraphsForDeltaChannels(collectionID, vDeltaChannels)
+	channel2FlowGraph, err := w.node.dataSyncService.addFlowGraphsForDeltaChannels(collectionID, vDeltaChannels)
+	if err != nil {
+		log.Warn("watchDeltaChannel, add flowGraph for deltaChannel failed", zap.Int64("collectionID", collectionID), zap.Strings("vDeltaChannels", vDeltaChannels), zap.Error(err))
+		return err
+	}
 	consumeSubName := funcutil.GenChannelSubName(Params.QueryNodeCfg.MsgChannelSubName, collectionID, Params.QueryNodeCfg.QueryNodeID)
 	// channels as consumer
 	for _, channel := range vDeltaChannels {
 		fg := channel2FlowGraph[channel]
```
Comment (xige-16) [inline internal/querynode/task.go:539]: If the channel has already been watched, it may be panic here.

**Comment by jennyli-z** [comment]
PR-15317-20220126-0c5c2b5a7 manually run passed on qa cluster once

**Comment by xiaofan-luan** [comment]
please rebase with master and rerun

**Comment by XuanYang-cn** [comment]
@bigsheeper Are you planning to fix this conflict and merge this PR into master?

**Comment by bigsheeper** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
@bigsheeper Please rebase the latest master to fix the UT error.

**Review by XuanYang-cn — COMMENTED** [review]
/lgtm

**Pull request merged**